modules/apps/portal/portal-util-test/src/testIntegration/java/com/liferay/portal/util/mail/test/MailServiceTest.java

```diff
@@ -147,6 +147,40 @@ public void testGetSessionWithCompanyId() throws Exception {
 		}
 	}
 
+	@Test
+	public void testMailSessionRefreshesOnConfigurationUpdate()
+		throws Exception {
+
+		long companyId = RandomTestUtil.randomLong();
+		int port1 = 1111;
+		int port2 = 2222;
+
+		_updateAndAssertSMTPPort(companyId, port1);
+
+		_updateAndAssertSMTPPort(companyId, port2);
+
+		_mailService.clearSession();
+	}
+
+	private void _updateAndAssertSMTPPort(long companyId, int port)
+		throws Exception {
+
+		_configurationProvider.saveCompanyConfiguration(
+			companyId,
+			"com.liferay.mail.settings.configuration." +
+				"MailSettingCompanyConfiguration",
+			HashMapDictionaryBuilder.<String, Object>put(
+				"outgoingSMTPPort", port
+			).build());
+
+		Thread.sleep(1000);
```

Comment:
when asserting port 2, it did not wait until the session is cleared. 
I can add a listener to wait till the configuraiton is updated, but I cannot find a way to listen until the session is clear.